mobile-ui/src/main/java/com/playone/mobile/ui/model/LoginViewModel.kt

```diff
@@ -1,10 +1,69 @@
 package com.playone.mobile.ui.model
 
+import android.arch.lifecycle.LifecycleObserver
+import android.arch.lifecycle.MutableLiveData
 import android.arch.lifecycle.ViewModel
 import android.arch.lifecycle.ViewModelProvider
+import android.os.Handler
+import androidx.os.postDelayed
+import com.playone.mobile.presentation.ViewResponse
 import com.playone.mobile.presentation.login.LoginPlayoneContract
+import com.playone.mobile.presentation.model.UserView
+import com.playone.mobile.ui.ext.ifFalse
+import com.playone.mobile.ui.ext.ifTrue
 
-class LoginViewModel(private val presenter: LoginPlayoneContract.Presenter) : ViewModel() {
+class LoginViewModel(private var presenter: LoginPlayoneContract.Presenter)
+    : ViewModel(), LifecycleObserver, LoginPlayoneContract.View {
+
+    val isProgressing: MutableLiveData<Boolean> = MutableLiveData()
+    val isSignedIn: MutableLiveData<Boolean> = MutableLiveData()
+    val occurredError: MutableLiveData<Throwable> = MutableLiveData()
+
+    init {
+        presenter.setView(this)
+    }
+
+    override fun setPresenter(presenter: LoginPlayoneContract.Presenter) {
+        this.presenter = presenter
+        this.presenter.setView(this)
+    }
+
+    override fun onResponse(response: ViewResponse<UserView>) {
+        when(response.status) {
+            ViewResponse.Status.LOADING -> { isProgressing.value = true }
+            ViewResponse.Status.ERROR -> {
+                isProgressing.value = false
+                occurredError.value = response.error
+            }
+            ViewResponse.Status.SUCCESS -> {
+                isProgressing.value = false
+                isSignedIn.value = true
+            }
+        }
+    }
+
+    fun signIn(email: String, password: String) {
+        // TODO: Need Email format verification
+        (email.isEmpty() or password.isEmpty()).ifTrue {
```

Comment:
`and`, `or`, `xor`, `not` I think they ought to be used in the operation rather than in the statement.

Reply:
I am not sure what you actually want to say but In my opinion, they are not related... 
statement can be included operator, assignment, and conditions. For me the statement above is not a problem...